MetaMask/metamask-extension
# Sign-in with Ethereum Domain Binding

## Explanation
This PR prevents Sign-in with Ethereum messages being served from a domain that doesn't match the domain in message from being displayed to the user.

## Manual Testing Steps

1. Turn on the feature flag by adding SIWE_V1=1 to metamask-extension/.metamaskrc
2. Build and run this version of the app.
3. Go to Metamask's [Test Dapp](https://metamask.github.io/test-dapp/) and scroll down to the Sign-In with Ethereum section. (If the dapp doesn't have this https://github.com/MetaMask/test-dapp/pull/164, [use this version](https://spruceid.github.io/test-dapp/)).
4. Test the 5 tests cases.
  i. The `Sign In With Ethereum (Bad Domain)` should show an error in the console. The rest of the tests should show the updated Sign-in with Ethereum display.
